File: Pcl_P2/TestHashTables.pas
PROGRAM TestHashTables;

CONST
    TEN = 10;

TYPE
    greek = (ALPHA, BETA, GAMMA);
    days = (MON, TUE, WED, THU, FRI, SAT, SUN);
    weekdays = MON..FRI;

    { Three proper named hashtable definitions }
    ht1 = HASHTABLE <integer, boolean>;
    ht2 = HASHTABLE <greek, integer>;
    ht3 = HASHTABLE <char, HASHTABLE <boolean, real>>;

VAR
    h1 : ht1;
    h2 : ht2;
    h3 : ht3;

    { Three proper unnamed hashtable definitions }
    u1 : HASHTABLE <1..TEN, real>;
    u2 : HASHTABLE <(RED, GREEN, BLUE), boolean>;
    u3 : HASHTABLE <weekdays, greek>;

BEGIN
END.

File: TestHashTables.java
import java.util.HashMap;

public class TestHashTables
{
    private static final int TEN = 10;

    private static enum greek {ALPHA, BETA, GAMMA};
    private static enum days {MON, TUE, WED, THU, FRI, SAT, SUN};

    private static HashMap<Integer, Boolean> h1 = new HashMap<>();
    private static HashMap<greek, Integer> h2 = new HashMap<>();
    private static HashMap<Character, HashMap<Boolean, Double>> h3 = new HashMap<>();
    private static HashMap<Integer, Double> u1 = new HashMap<>();
    private static HashMap<Integer, Boolean> u2 = new HashMap<>();
    private static HashMap<days, greek> u3 = new HashMap<>();

    public static void main(String[] args)
    {
        System.out.println("Done!");
    }
}
